### ISSUE TITLE
Port twisted.internet.defer to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5947]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5947|
|Type|enhancement|
|Created|2012-09-06 14:59:49Z|
|Branch|https://github.com/twisted/twisted/tree/defer-py3-5947|

`defer.py` should run under Python 3.3.

<details><summary>Searchable metadata</summary>

```
trac-id__5947 5947
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_defer_py3_5947 branches/defer-py3-5947
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346943589000000 1346943589000000
changetime__1347482164000000 1347482164000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

As part of this ticket we should make sure to re-enable the `inlineCallbacks` test for `failure.Failure` under Python 3.3.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35589]) Branching to 'defer-py3-5947'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/defer-py3-5947 will run eventually.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. Some future imports needed on defer.py
  1. in `failure.py`, `test_cleanFailureRemovesTracebackInPython3` could use `getDivisionFailure`, I think.

Seems good otherwise.  Please fix these points and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [35620]) Merge defer-py3-5947

Author: itamar
Review: exarkun
Fixes: [#5947](https://github.com/twisted/twisted/issues/5947)

Port twisted.internet.defer to Python 3.
